### PR TITLE
Adds excludeScoped feature

### PR DESCRIPTION
**Pull request by jrummler**
Hi!

I added a feature to exclude scopes from license check. We do not want our test dependencies checked, because they will not be used in production or shipped. You could also use this to avoid check of libraries with scope "provided".

Best regards!!
Jens

### CODE REVIEW

File: src/main/java/org/complykit/licensecheck/mojo/OpenSourceLicenseCheckMojo.java [hunk 1]
```diff
@@ -238,6 +244,10 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
     }
 
+    private String toCoordinates(Artifact artifact) {
```
Comment (mrice) [inline src/main/java/org/complykit/licensecheck/mojo/OpenSourceLicenseCheckMojo.java]: good idea!

File: src/main/java/org/complykit/licensecheck/mojo/OpenSourceLicenseCheckMojo.java [hunk 2]
```diff
@@ -428,7 +438,11 @@ private void loadDescriptors() {
         }
     }
 
-    private boolean isExcluded(final Set<String> excludeSet, final List<Pattern> patternList, final String template) {
+    private boolean isExcluded(final Set<String> excludeSet, final List<Pattern> patternList, final Set<String> excludedScopes, final Artifact artifact){
+        return isExcludedTemplate(excludeSet, patternList, toCoordinates(artifact)) || excludedScopes.contains(artifact.getScope());
+    }
+
+    private boolean isExcludedTemplate(final Set<String> excludeSet, final List<Pattern> patternList, final String template) {
```
Comment (mrice) [inline src/main/java/org/complykit/licensecheck/mojo/OpenSourceLicenseCheckMojo.java]: bueno

File: src/main/java/org/complykit/licensecheck/mojo/OpenSourceLicenseCheckMojo.java [hunk 3]
```diff
@@ -127,6 +127,13 @@ of this software and associated documentation files (the "Software"), to deal
     private String[] blacklist;
 
     /**
+     * A list of scopes to exclude. May be used to exclude artifacts with test or provided scope from license check.
+     * Example: &lt;configuration&gt; &lt;excludedScopes&gt; &lt;param&gt;test&lt;/param&gt; &lt;param&gt;provided&lt;/param&gt; &lt;/excludedScopes&gt; &lt;/configuration&gt;
+     */
```
Comment (mrice) [inline src/main/java/org/complykit/licensecheck/mojo/OpenSourceLicenseCheckMojo.java]: nice, thanks!

**Comment by mrice** [comment]
Awesome!! Thanks so much, but of course now I have to dig out my old release instructions ;-) I'll try to cut a new release tomorrow. thanks again

**Comment by jrummler** [comment]
Take your time! Thanks for merging & releasing! ;-)

**Comment by jrummler** [comment]
Hi! 
Still waiting for a release - will there be one any time soon?
Thank you!
Jens

**Comment by mrice** [comment]
Oh crap… Sorry about this. Not sure what I can commit to… any ideas? Can I let you do it somehow?

**Comment by jrummler** [comment]
Sure I could do it - but I don't know if you want me to:
- be able to commit into your master (my GitHub account)
- be able to release new Maven artefacts in org.complykit (my Sonatype accout)

I think it would be a lot easier if you would run a release build.

All that is necessary should be:
mvn release:prepare release:perform -Psonatype-oss-release

Then login to Sonatype central repo and close & release the staging repo.

Thank you!
Jens

**Comment by mrice** [comment]
@jrummler just for the record... as absurdley late as this is, I *am* working on this over the labor day weekend

**Comment by mrice** [comment]
@jrummler hey if you have a few free moments this weekend, can you check the readme to see if I stated your changes correctly?

**Comment by mrice** [comment]
Ok released 0.5.3, but still want to know what you think of the readme updates

**Comment by jrummler** [comment]
Hey! Thanks a lot.
There is a typo in the headline (score instead of scope) and I would prefer not to be mentioned in the readme. Maybe instead it would be great to explain why it makes sense to exclude these scopes - you don't ship them with your code. Maybe this clarifies for some users why you want to set these excludes!

Thank you for this great plugin! And, just for the record, we don't have labor day here in Germany! ;-)

**Pull request merged**